core/src/main/java/io/grpc/StreamTracer.java

```diff
@@ -36,19 +36,65 @@ public void streamClosed(Status status) {
    * An outbound message has been passed to the stream.  This is called as soon as the stream knows
    * about the message, but doesn't have further guarantee such as whether the message is serialized
    * or not.
+   *
+   * @deprecated use {@link #outboundMessage(int)}
    */
+  @Deprecated
   public void outboundMessage() {
   }
 
   /**
+   * An outbound message has been passed to the stream.  This is called as soon as the stream knows
+   * about the message, but doesn't have further guarantee such as whether the message is serialized
+   * or not.
+   *
+   * @param seqNo the sequential number of the message within the stream, starting from 0
+   */
+  public void outboundMessage(int seqNo) {
```

Comment:
Why is `seqNo` necessary? Wouldn't it be easy for the implementation to do `i++`? Or is this to make clear the correlation between outboundMessage events and the respective outboundMessageSent event?

Reply:
The main reason is to make the correlation clear.

Reply:
Make a note of that in the Javadoc.